kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/LogReplay.java

```diff
@@ -243,6 +243,12 @@ private void verifySupportedColumnMappingMode(Metadata metadata) {
      * Verifies that a set of delta or checkpoint files to be read actually belongs to this table.
      */
     private static void assertLogFilesBelongToTable(Path logPath, List<FileStatus> allFiles) {
-        // TODO:
+        for (FileStatus fileStatus : allFiles) {
+            String filePath = fileStatus.getPath();
+            if (!new Path(filePath).getParent().equals(new Path(logPath.toUri()))) {
```

Comment:
We are planning to remove the `Path` as the code is copied from the Hadoop and we don't want to maintain a copy.

Wondering can you directly use the prefix search? Both the `logPath` and `FileStatus.getPath` are fully resolved.

Reply:
Thanks for the feedback. To clarify, are you saying that the plan is to remove the `Path` class entirely? What will the logPath parameter type change to?

Reply:
Yep you are right, it is passed around a lot, but created in just few places. 

For now keep the `Path` as input arg and use the `path.toUri.toString` (create once in the beginning) for comparison.

Reply:
Should be resolved, we will see if tests pass